api/v1/cluster_webhook.go

```diff
@@ -143,6 +144,17 @@ func (r *Cluster) setDefaults(preserveUserSettings bool) {
 	if !r.Spec.Monitoring.AreDefaultQueriesDisabled() {
 		r.defaultMonitoringQueries(configuration.Current)
 	}
+
+	// If the API or validation webhooks are disable we set the replicationSlots to true if nil
+	if r.Spec.ReplicationSlots == nil {
+		r.Spec.ReplicationSlots = &ReplicationSlotsConfiguration{}
+	}
+	if r.Spec.ReplicationSlots.HighAvailability == nil {
+		r.Spec.ReplicationSlots.HighAvailability = &ReplicationSlotsHAConfiguration{
+			Enabled:    ptr.To(true),
+			SlotPrefix: "_cnpg_",
+		}
+	}
```

Comment:
The logic here is a bit obscure. Line 149... we create an empty ReplicationSlots config. Line 152 WILL get executed right after, and will add the replication slots by default.
Why not merge the two? Or am I missing something?